From 1f88dec0e57d9cbcea0ee53da600a022f5297b58 Mon Sep 17 00:00:00 2001
From: manigandanj <47597704+manigandanj@users.noreply.github.com>
Date: Fri, 27 Mar 2026 21:21:51 +0530
Subject: [PATCH] =?UTF-8?q?RDKCOM-5553:=20RDKBDEV-3401,=20RDKBACCL-1526=20?=
 =?UTF-8?q?RNDIS=20interface=20is=20not=20working=20when=20DHCPManager=20i?=
 =?UTF-8?q?s=20ena=E2=80=A6=20(#85)?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

…bled

Reason for change: MTU is set to 0, so usb0 interface is not having IP
Test procedure: Able to ping Internet through usb0
Risks: Low

Signed-off-by: Manigandan Gopalakrishnan <Manigandan_Gopalakrishnan@comcast.com>
---
 source/DHCPMgrUtils/dhcpmgr_v4_lease_handler.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/source/DHCPMgrUtils/dhcpmgr_v4_lease_handler.c b/source/DHCPMgrUtils/dhcpmgr_v4_lease_handler.c
index 7acb052..ad71897 100644
--- a/source/DHCPMgrUtils/dhcpmgr_v4_lease_handler.c
+++ b/source/DHCPMgrUtils/dhcpmgr_v4_lease_handler.c
@@ -356,6 +356,9 @@ void DhcpMgr_ProcessV4Lease(PCOSA_DML_DHCPC_FULL pDhcpc)
                 DhcpMgr_PublishDhcpV4Event(pDhcpc, DHCP_LEASE_DEL);
                 continue;
             }
+            //If DHCP server is not sending MTU, setting the default value of 1500. Can't set 0 as MTU
+            if(newLease->mtuSize == 0)
+                newLease->mtuSize = 1500;
             DHCPMGR_LOG_INFO("%s %d: NewLease->address %s  \n",__FUNCTION__, __LINE__, newLease->address);
             DHCPMGR_LOG_INFO("%s %d: NewLease->netmask %s  \n",__FUNCTION__, __LINE__, newLease->netmask);
             DHCPMGR_LOG_INFO("%s %d: NewLease->gateway %s  \n",__FUNCTION__, __LINE__, newLease->gateway );
